embassy-nrf/src/gpio.rs

```diff
@@ -205,6 +166,101 @@ impl<'d, T: Pin> StatefulOutputPin for Output<'d, T> {
     }
 }
 
+/// GPIO flexible pin.
+///
+/// This pin can either be a disconnected, input, or output pin.
+pub struct FlexPin<'d, T: Pin> {
+    pub(crate) pin: T,
+    phantom: PhantomData<&'d mut T>,
+}
+
+impl<'d, T: Pin> FlexPin<'d, T> {
+    /// Wrap the pin in a `FlexPin`.
+    ///
+    /// The pin remains disconnected.
+    pub fn new(pin: impl Unborrow<Target = T> + 'd) -> Self {
+        unborrow!(pin);
+        // Pin will be in disconnected state.
+        Self {
+            pin,
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn to_input(&self, pull: Pull) {
```

Comment:
On naming: 
- `to_whatever` sounds like it returns a new struct of the new type
- `disconnect` is inconsistent.

What about `set_as_input`, `set_as_output`, `set_as_disconnected`?

Reply:
Yeah I like your names more.